.github/workflows/templates/configurations.yml

```diff
@@ -3,25 +3,25 @@
 # - config: a string encoded list containing the docker image name and any number of build options.
 # - The build options are `mpi`, `tbb`, `llvm`, `nomd`, and `nohpmc`.
 unit_test_configurations:
-- config: "[clang14_py310, mpi, tbb, llvm]"
-- config: "[gcc12_py310]"
-- config: "[gcc12_py310, nomd]"
-- config: "[gcc12_py310, nohpmc]"
-- config: "[gcc12_py310, nomd, nohpmc]"
-- config: "[cuda117_gcc11_py310, mpi, llvm, debug]"
-- config: "[cuda117_gcc11_py310, mpi, llvm]"
-- config: "[cuda117_gcc11_py310]"
+- config: "[clang14_py311, mpi, tbb, llvm]"
+- config: "[gcc12_py311]"
+- config: "[gcc12_py311, nomd]"
+- config: "[gcc12_py311, nohpmc]"
+- config: "[gcc12_py311, nomd, nohpmc]"
+- config: "[cuda118_gcc11_py310, mpi, llvm, debug]"
+- config: "[cuda118_gcc11_py310, mpi, llvm]"
+- config: "[cuda118_gcc11_py310]"
 - config: "[gcc9_py39]"
 - config: "[gcc7_py37]"
 - config: "[gcc7_py36]"
 
 # Configurations on which to run longer validation tests. Must be a subset of
 # `unit_test_configurations`
 validate_configurations:
-- config: "[clang14_py310, mpi, tbb, llvm]"
-- config: "[gcc12_py310]"
-- config: "[cuda117_gcc11_py310, mpi, llvm]"
-- config: "[cuda117_gcc11_py310]"
+- config: "[clang14_py311, mpi, tbb, llvm]"
+- config: "[gcc12_py311]"
+- config: "[cuda118_gcc11_py310, mpi, llvm]"
+- config: "[cuda118_gcc11_py310]"
```

Comment:
Is cuda 11.8 compatible with gcc 12? If so, it might be a good idea to have at least one configuration with the newest versions of cuda, python, and gcc.

Reply:
I don't know if CUDA 11.8 is compatible with gcc 12. 

I use NVIDIA provided CUDA development images (https://hub.docker.com/r/nvidia/cuda/tags) which install CUDA into an Ubuntu with the *default* compiler (gcc10 in this case). It would take some careful package management in https://github.com/glotzerlab/ci-containers to switch these containers to a different gcc and that is not an effort that I have made.

Reply:
How do you test in environments with cuda11.8 and gcc11 then?

Reply:
gcc11 is default in ubuntu 22.04. gcc9 is default in ubuntu 20.04. I mistyped when I mentioned gcc10 was default.
NVIDIA provides CUDA docker images with different ubuntu LTS versions as the base: https://hub.docker.com/r/nvidia/cuda/tags. I typically choose the latest available at the time I build the images.